src/shogun/clustering/Hierarchical.cpp

```diff
@@ -212,8 +217,83 @@ SGMatrix<int32_t> CHierarchical::get_cluster_pairs()
 	return SGMatrix<int32_t>(pairs,2,merges, false);
 }
 
-
 void CHierarchical::store_model_features()
 {
-	/* TODO. Currently does nothing since apply methods are not implemented. */
+	CDenseFeatures<float64_t>* rhs = 
+		(distance->get_rhs())->as<CDenseFeatures<float64_t>>();
+
+	int32_t num_vectors = rhs->get_num_vectors();
+	int32_t num_features = rhs->get_num_features();
+
+	SGVector<int32_t> n_cluster_samples = SGVector<int32_t>(
+		num_vectors + merges);
+	shogun::linalg::set_const(n_cluster_samples, 0);
+
+	SGMatrix<float64_t> null_matrix = SGMatrix<float64_t>(
+		num_features, num_vectors + merges);
+	CDenseFeatures<float64_t>* temp_cluster_centers =
+	    new CDenseFeatures<float64_t>(null_matrix);
+
+	for (int32_t i = 0; i < num_vectors; i++)
+	{
+		bool dofree_c, dofree_rhs;
+		int32_t centerIdx = assignment[i];
+		SG_DEBUG("\n");
+		SG_DEBUG("On %04i point, with assignment=%04i \n", i, centerIdx);
+		float64_t* center = temp_cluster_centers->get_feature_vector(
+		    centerIdx, num_features, dofree_c);
+		float64_t* sample_point =
+		    rhs->get_feature_vector(i, num_features, dofree_rhs);
+		for (int32_t j = 0; j < num_features; j++)
+		{
+			if (n_cluster_samples[centerIdx] > 0)
+			{
+				center[j] = (center[j] * n_cluster_samples[centerIdx] +
```

Comment:
you should leave a comment on what type of algorithm/approach is used here to store/compute model features